physx/src/transform.rs

```diff
@@ -48,70 +40,45 @@ impl Into<PxQuat> for PxQuatWrap {
-    fn from(other: Isometry3<f32>) -> Self {
-        let quat: glm::Quat = glm::convert(other.rotation);
-        let quaternion: PxQuat = PxQuatWrap::from(quat).into();
-        let trans = other.translation;
-        PxTransformWrap(unsafe { PxTransform_new_4(trans.x, trans.y, trans.z, quaternion) })
-    }
+/// Convert a Mat4 to a PxTransform
+pub fn gl_to_px_tf(trans: Mat4) -> PxTransform {
+    gl_to_px_tf_ref(&trans)
 }
 
-impl Into<PxTransform> for PxTransformWrap {
-    /// Unwrap helper
-    fn into(self) -> PxTransform {
-        self.0
-    }
+/// Convert a reference to a Mat4 to a PxTransform
+pub fn gl_to_px_tf_ref(trans: &Mat4) -> PxTransform {
+    let quat: PxQuatWrap = Quat::from_rotation_mat4(&trans).into();
+    let trans = trans.w_axis().truncate();
+    unsafe { PxTransform_new_4(trans.x(), trans.y(), trans.z(), quat.0) }
 }
 
-/// Convert a glm::Mat4 to a PxTransform
-pub fn na_to_px_tf(trans: glm::Mat4) -> PxTransform {
-    let iso: Isometry3<f32> = glm::try_convert(trans).expect("non isometric transform");
-
-    PxTransformWrap::from(iso).into()
-}
-
-/// Convert a reference to a glm::Mat4 to a PxTransform
-pub fn na_to_px_tf_ref(trans: &glm::Mat4) -> PxTransform {
-    let iso: Isometry3<f32> = glm::try_convert(*trans).expect("non isometric transform");
-
-    PxTransformWrap::from(iso).into()
-}
-
-/// Convert a PxTransform to a glm::Mat4
-pub fn px_to_na_tf(trans: PxTransform) -> glm::Mat4 {
-    let translation = px_to_na_v3(trans.p);
-    let quaternion = px_to_na_q(trans.q);
-
-    nalgebra::Isometry3::from_parts(
-        translation.into(),
-        nalgebra::UnitQuaternion::new_normalize(quaternion),
-    )
-    .into()
+/// Convert a PxTransform to a Mat4
+pub fn px_to_gl_tf(trans: PxTransform) -> Mat4 {
+    let translation = px_to_gl_v3(trans.p);
+    let quaternion = px_to_gl_q(trans.q);
+    Mat4::from_rotation_translation(quaternion, translation)
 }
 
-/// Convert a PxQuat to a glm::Quat
+/// Convert a PxQuat to a Quat
 #[inline(always)]
-pub fn px_to_na_q(quat: PxQuat) -> glm::Quat {
-    glm::quat(quat.x, quat.y, quat.z, quat.w)
+pub fn px_to_gl_q(quat: PxQuat) -> Quat {
+    Quat::new(quat.x, quat.y, quat.z, quat.w)
 }
 
-/// Convert a PxQuat to a glm::Quat
+/// Convert a PxQuat to a Quat
 #[inline(always)]
-pub fn na_to_px_q(quat: glm::Quat) -> PxQuat {
+pub fn gl_to_px_q(quat: Quat) -> PxQuat {
     PxQuatWrap::from(quat).into()
 }
 
-/// Convert a PxVec3 to a glm::Vec3
+/// Convert a PxVec3 to a Vec3
 #[inline(always)]
-pub fn px_to_na_v3(pos: PxVec3) -> glm::Vec3 {
-    glm::vec3(pos.x, pos.y, pos.z)
+pub fn px_to_gl_v3(pos: PxVec3) -> Vec3 {
+    Vec3::new(pos.x, pos.y, pos.z)
 }
 
-/// Convert a PxVec3 to a glm::Vec3
+/// Convert a PxVec3 to a Vec3
 #[inline(always)]
-pub fn na_to_px_v3(pos: glm::Vec3) -> PxVec3 {
-    unsafe { PxVec3_new_3(pos.x, pos.y, pos.z) }
+pub fn gl_to_px_v3(pos: Vec3) -> PxVec3 {
+    unsafe { PxVec3_new_3(pos.x(), pos.y(), pos.z()) }
```

Comment:
Might be better to do `let (x, y, z) = pos.into(); ` rather than accessing each component separately.